web/src/engine/attachment/src/pageContextAttachment.ts

```diff
@@ -0,0 +1,1130 @@
+   * Scope        Private
+   * @param       {Element}      Pelem       Element to be disabled
+   * Description  Disable KMW control element
+   */
+  _DisableControl(Pelem: HTMLElement) {
+    // Only operate on attached elements!  Non-design-mode IFrames don't get attachment markers, so we check them specifically instead.
+    if(this.isAttached(Pelem) || Pelem instanceof Pelem.ownerDocument.defaultView.HTMLIFrameElement) {
+      if(this.device.touchable) {
+        this.disableTouchElement(Pelem);
+      }
+
+      this.listInputs(); // Fix up our internal input ordering scheme.
+      this.disableInputElement(Pelem);
+    }
+  }
+
+  /**
+   * Function     _EnableControl
+   * Scope        Private
+   * @param       {Element}    Pelem   Element to be enabled
+   * Description  Enable KMW control element
+   */
+  _EnableControl(Pelem: HTMLElement) {
+    if(this.isAttached(Pelem)) { // Only operate on attached elements!
+      if(this.device.touchable) {
+        this.enableTouchElement(Pelem);
+
+      } else {
+        this.enableInputElement(Pelem);
+      }
+
+      this.listInputs();
+    }
+  }
+
+
+  /**
+   * Function     disableControl
+   * Scope        Public
+   * @param       {Element}      Pelem       Element to be disabled
+   * Description  Disables a KMW control element
+   */
+  disableControl(Pelem: HTMLElement) {
+    if(!this.isAttached(Pelem)) {
+      console.warn("KeymanWeb is not attached to element " + Pelem);
+    }
+
+    var cn = Pelem.className;
+    if(cn.indexOf('kmw-disabled') < 0) { // if not already explicitly disabled...
+      Pelem.className = cn ? cn + ' kmw-disabled' : 'kmw-disabled';
+    }
+
+    // The rest is triggered within MutationObserver code.
+    // See _EnablementMutationObserverCore.
+  }
+
+  /**
+   * Function     enableControl
+   * Scope        Public
+   * @param       {Element}      Pelem       Element to be disabled
+   * Description  Disables a KMW control element
+   */
+  enableControl(Pelem: HTMLElement) {
+    if(!this.isAttached(Pelem)) {
+      console.warn("KeymanWeb is not attached to element " + Pelem);
+    }
+
+    var cn = Pelem.className;
+    var tagIndex = cn.indexOf('kmw-disabled');
+    if(tagIndex >= 0) { // if already explicitly disabled...
+      Pelem.className = cn.replace('kmw-disabled', '').trim();
+    }
+
+    // The rest is triggered within MutationObserver code.
+    // See _EnablementMutationObserverCore.
+  }
+
+  // Create an ordered list of all text and search input elements and textarea elements
+  // except any tagged with class 'kmw-disabled'
+  // TODO: email and url types should perhaps use default keyboard only
+  listInputs() {
+    let eList: SortableInput[]=[];
+    let t1=document.getElementsByTagName('input');
+    let t2=document.getElementsByTagName('textarea');
+
+    for(let i=0; i<t1.length; i++) {
+      switch(t1[i].type) {
+        case 'text':
+        case 'search':
+        case 'email':
+        case 'url':
```

Comment:
This set is different to `isKMWInput` test.

Reply:
Guess that should be marked as an issue?   Or should I go ahead and fix it here?  And, if the latter, should I use the larger set?

From `master`:

https://github.com/keymanapp/keyman/blob/fafe498775d4a01445e44fb92e59e7428c5f80f8/web/src/engine/main/dom/domManager.ts#L805

https://github.com/keymanapp/keyman/blob/fafe498775d4a01445e44fb92e59e7428c5f80f8/web/src/engine/main/dom/domManager.ts#L812-L823

and...

https://github.com/keymanapp/keyman/blob/fafe498775d4a01445e44fb92e59e7428c5f80f8/web/src/engine/main/dom/domManager.ts#L495-L501

Reply:
> Guess that should be marked as an issue?

Fortunately you answered your own question at #8848 before I could get to it :grin: